webpack/assets/javascripts/react_app/common/hooks/useSubmitOnEnter.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+/**
+ * A custom hook that listens for enter on a ref and runs a function on enter keydown
+ * @param  {object} ref a ref created with React's useRef or createRef that scopes the event listener
+ * @param  {function} onSubmit function to execute on enter keydown
+ */
+export const useSubmitOnEnter = (ref, onSubmit) => {
+  const listener = event => {
+    if (event.code === 'Enter' || event.code === 'NumpadEnter') onSubmit();
+  };
+
+  return useEffect(() => {
+    const refCopy = ref; // re-assigning so it's available on unmount
```

Comment:
Since you're assigning it with `=` and not `Object.assign`, isn't it pointing to the same object?  maybe call it `refRef` or something :laughing:

Reply:
I changed it to `refPointer`, which is probably a more accurate name, but open to other suggestions :smile: